internal/service/advancedcluster/resource_advanced_cluster_migration_test.go

```diff
@@ -47,7 +52,89 @@ func TestMigAdvancedCluster_replicaSetMultiCloud(t *testing.T) {
 			{
 				ExternalProviders: mig.ExternalProviders(),
 				Config:            config,
-				Check:             checkReplicaSetMultiCloud(clusterName, 3),
+				Check:             checkReplicaSetMultiCloud(clusterName, 3, false),
+			},
+			mig.TestStepCheckEmptyPlan(config),
+			{
+				ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
+				Config:                   config,
+				Check:                    checkReplicaSetMultiCloud(clusterName, 3, true), // external_id will be present in latest version
+			},
+		},
+	})
+}
+
+func TestMigAdvancedCluster_singleShardedMultiCloud(t *testing.T) {
+	var (
+		orgID       = os.Getenv("MONGODB_ATLAS_ORG_ID")
+		projectName = acc.RandomProjectName() // No ProjectIDExecution to avoid cross-region limits because multi-region
+		clusterName = acc.RandomClusterName()
+		config      = configSingleShardedMultiCloud(orgID, projectName, clusterName)
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { mig.PreCheckBasic(t) },
+		CheckDestroy: acc.CheckDestroyCluster,
+		Steps: []resource.TestStep{
+			{
+				ExternalProviders: mig.ExternalProviders(),
+				Config:            config,
+				Check:             checkSingleShardedMultiCloud(clusterName, false),
+			},
+			mig.TestStepCheckEmptyPlan(config),
+			{
+				ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
+				Config:                   config,
+				Check:                    checkSingleShardedMultiCloud(clusterName, true), // external_id will be present in latest version
+			},
+		},
+	})
+}
+
+func TestMigAdvancedCluster_singleShardPerZoneGeoSharded(t *testing.T) {
+	var (
+		orgID       = os.Getenv("MONGODB_ATLAS_ORG_ID")
+		projectName = acc.RandomProjectName() // No ProjectIDExecution to avoid cross-region limits because multi-region
+		clusterName = acc.RandomClusterName()
+		config      = configGeoShardedOldSchema(orgID, projectName, clusterName, 1, 1, false)
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { mig.PreCheckBasic(t) },
+		CheckDestroy: acc.CheckDestroyCluster,
+		Steps: []resource.TestStep{
+			{
+				ExternalProviders: mig.ExternalProviders(),
+				Config:            config,
+				Check:             checkGeoShardedOldSchema(clusterName, 1, 1, false, false),
+			},
+			mig.TestStepCheckEmptyPlan(config),
+			{
+				ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
+				Config:                   config,
+				Check:                    checkGeoShardedOldSchema(clusterName, 1, 1, true, true), // external_id will be present in latest version
+			},
+		},
+	})
+}
+
+func TestMigAdvancedCluster_symmetricGeoShardedOldSchema(t *testing.T) {
+	acc.SkipTestForCI(t) // TODO: CLOUDP-260154 for ensuring old schema is supported
```

Comment:
is this comment still valid?

Reply:
yes still valid, the test is currently failing and we have a fix pending from API side.